frontend-react/src/contexts/SessionStorageContext.tsx

```diff
@@ -0,0 +1,34 @@
+import { createContext, FC, useMemo } from "react";
+
+import useSessionStorage, { SessionStore } from "../hooks/UseSessionStorage";
+
+interface ISessionContext {
+    values: SessionStore;
+    updateSessionStorage: (store: Partial<SessionStore>) => void;
+}
+
+export const SessionStorageContext = createContext<ISessionContext>({
+    values: {},
+    updateSessionStorage: (store: Partial<SessionStore>) => {
+        console.log("to please SonarCloud");
```

Comment:
Contexts require some initial value that matches the shape of the payload, and the update functions return void since they're just updating state. Leaving those brackets empty, though, results in a SonarCloud smell, and making the update function nullable means having to check it _every...single...time..._ that you wanna use it across all those files.

Reply:
So, maybe make it optional and just leave it out?

```typescript
export interface SessionController {
    values: SessionStore;
    updateSessionStorage?: (p: Partial<SessionStore>) => void;
}
//...
export const SessionStorageContext = createContext<SessionController>({
    values: {},
 });

```

Reply:
I don't want to have to check if it's truthy every time we use it, though. And SonarCloud hates when I `trustMeItsNotNull!!` it, haha

I wonder if something like this would satisfy both SonarCloud and our linter:
```typescript
export const SessionStorageContext = createContext<SessionController>({
    values: {},
    updateSessionStorage: (p: Partial<SessionStore>) => { return void }
 });
```

Reply:
It is also worth noting that the `console.log()` is never actually called! That context default never gets used anywhere, when you call `useContext(SessionStorageContext)`, you're tapping into the value in `<SessionStorageContext.Provider>` which is always the state and updater function.

Reply:
Since it never should be called you could change the console.log to `console.error("Should not be called for this component")`? Then I'll be happy.